### PR TITLE
Implement statsbeat shutdown

### PR DESCRIPTION
**Pull request by lzchen**

### CODE REVIEW

File: sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/_base.py [hunk 1]
```diff
@@ -236,19 +248,32 @@ def _transmit(self, envelopes: List[TelemetryItem]) -> ExportResult:
...
                     _update_requests_map('count')
                     _update_requests_map(_REQ_DURATION_NAME[1], value=end_time-start_time)
+                if self._is_statsbeat_initializing_state():
+                    # Update statsbeat initial success state if reached ingestion
+                    if reach_ingestion:
+                        set_statsbeat_initial_success(True)
+                    else:
+                        # if didn't reach ingestion, increment and check if failure threshold
+                        # has been reached during attempting statsbeat initialization
+                        if increment_and_check_statsbeat_failure_count():
```
Comment (jeremydvoss) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/_base.py:271]: Do we only shutdown based on failures upon start up, not afterwards?

File: sdk/monitor/azure-monitor-opentelemetry-exporter/tests/statsbeat/test_statsbeat.py [hunk 2]
```diff
@@ -534,6 +545,23 @@ def test_get_feature_metric_none(self):
         self.assertEqual(len(observations), 0)
         self.assertEqual(_StatsbeatMetrics._FEATURE_ATTRIBUTES["feature"], 0)
 
+    # pylint: disable=protected-access
+    def test_get_feature_metric_instrumentation(self):
```
Comment (jeremydvoss) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/tests/statsbeat/test_statsbeat.py:549]: This is the same test name. Did you mean to have it be something else? It's a negative case.

File: sdk/monitor/azure-monitor-opentelemetry-exporter/tests/test_base_exporter.py [hunk 3]
```diff
@@ -215,22 +250,25 @@ def test_transmission_200(self):
         "APPLICATIONINSIGHTS_STATSBEAT_DISABLED_ALL": "false",
         }
     )
-    def test_statsbeat_200(self):
+    @mock.patch("azure.monitor.opentelemetry.exporter.statsbeat._statsbeat.collect_statsbeat_metrics")
+    def test_statsbeat_200(self, stats_mock):
```
Comment (jeremydvoss) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/tests/test_base_exporter.py:254]: Not familiar with this style. Is stats_mock the mock path of collect_statsbeat_metrics? That automatically adds it as an argument?

File: sdk/monitor/azure-monitor-opentelemetry-exporter/tests/statsbeat/test_statsbeat.py (hunk 2 above)
Comment (lzchen) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/tests/statsbeat/test_statsbeat.py:549]: Oops. I removed the previous one.

**Review by jeremydvoss — APPROVED** [review]

**Pull request merged**